### ISSUE TITLE
RFP: Clarification on Generic Subcommand Index Handling

### ISSUE THREAD

**Issue by qdot** [buttplugio/buttplug#64]
We don't currently have any rules about how we handle repeated subindexes in Generic Commands.

For instance, with VibrateCmd, say someone creates the following packet:

```json
{
"VibrateCmd":
{
"Speeds": [
{ "Index": 0, "Speed": 0.5 },
{ "Index": 0, "Speed": 0.75 }
]
}
"Id": 10,
"DeviceIndex": 0
}
```

Our implementations in both C# and JS would currently send 2 commands back to back, which is silly. We should specify whether either the last value for each index should be chosen, or whether we should error out on this case.

I don't have strong feelings either direction. Erroring out would require actual library logic though, I don't believe we'd easily be able to catch that via the schema.

**Comment by blackspherefollower** [comment]
It should probably be disallowed, implemented at the message handling layer within the drivers (since that's were we disallow too many and out of bounds subcommands).

Given that this logic is repeated so often, maybe a generic utility function should be written to handle these rules?

**Comment by blackspherefollower** [comment]
Ah, got it.
Then in my opinion, we should document that only one subcommand per feature should be sent.

**Comment by qdot** [comment]
Yup, only reason I was RFP'ing this in the first place is that we already shipped the message in January and I wanted to make sure I wasn't missing anything. I think we can refine the spec without rolling the version on this, since the logic will be required in the server instead of the schema anyways.

I'll update the spec tomorrow and land then, will close this with the commit message for it.
